### PR TITLE
feat(deploy):  Enables apps deployments with functions to environments

### PR DESCRIPTION
**Pull request by raulb**
## Description of change

Fixes https://github.com/meroxa/cli/issues/635
Fixes https://github.com/meroxa/cli/issues/637

This will be ready to ship once the following are done:

- [x] Shipped https://github.com/meroxa/platform-api/issues/1694
- [x] Shipped https://github.com/meroxa/meroxa-go/issues/162
- [x] `meroxa-go` updated to the its latest.

## Type of change

- [x]  New feature
- [ ]  Bug fix
- [ ]  Refactor
- [ ]  Documentation

## How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

## Demo

Assuming application contains a function.

**Before**

```
$ meroxa apps deploy --env my-env
Error: cannot deploy an application with a function to an environment
```

**After**

```
$ meroxa apps deploy --env my-env
...
```

## Additional references

⚠️ Blocked on https://github.com/meroxa/platform-api/issues/1694

### CODE REVIEW

File: cmd/meroxa/root/apps/deploy.go [hunk 1]
```diff
@@ -201,10 +202,20 @@ func (d *Deploy) Logger(logger log.Logger) {
 	d.logger = logger
 }
 
+// getAppSource will return the proper destination where the application source will be uploaded and fetched.
+func (d *Deploy) getAppSource(ctx context.Context) (*meroxa.Source, error) {
+	if env := d.flags.Environment; env != "" {
```
Comment (dianadoherty) [inline cmd/meroxa/root/apps/deploy.go:207]: I'm not sure that I agree about us using v2 only if customers have the feature flag enabled. I think the env info should be entirely handled by the API, whether customers use envs or common

Comment (raulb) [reply cmd/meroxa/root/apps/deploy.go:207]: @dianadoherty this is referring to the flag provided in the CLI (i.e.: `--env`), not the feature flag.

**Comment by anna-cross** [comment]
FYI, will be making changes to meroxa-go on Source object, so we might want to vendor here after

**Comment by raulb** [comment]
@anna-cross sounds good. There will still be some changes in both meroxa-go and CLI we'll need to make for builds, so I think that's fine.

**Review by anna-cross — APPROVED** [review]

**Pull request merged**